### PR TITLE
Use scheduler Profiles

### PR DESCRIPTION
**Pull request by damemi**
Fixes https://github.com/openshift/cluster-kube-scheduler-operator/issues/197

This updates the operator to parse a scheduler Profiles configmap and merge it with the config we already provide post-bootstrap

### CODE REVIEW

File: pkg/operator/configobservation/scheduler/observe_scheduler.go [hunk 1]
```diff
@@ -39,28 +42,67 @@ func ObserveSchedulerConfig(genericListers configobserver.Listers, recorder even
...
-	switch {
-	case len(configMapName) == 0:
-		sourceTargetLocation = resourcesynccontroller.ResourceLocation{}
-	case len(configMapName) > 0:
-		sourceTargetLocation = resourcesynccontroller.ResourceLocation{
-			Namespace: operatorclient.GlobalUserSpecifiedConfigNamespace,
-			Name:      configMapName,
+	// Deprecated handling for Policy config
+	// TODO(@damemi): Remove this when Policy API is removed
+	if len(schedulerConfig.Spec.Policy.Name) > 0 {
```
Comment (soltysh) [inline pkg/operator/configobservation/scheduler/observe_scheduler.go:48]: Maybe we should warn in the logs at least about this fact? Also, make sure to create an entry in the release notes for 4.7 mentioning the fact that we're switching from policy to profiles.

Comment (damemi) [reply pkg/operator/configobservation/scheduler/observe_scheduler.go:48]: Both good points, how do we make an entry in the release notes?

**Review by damemi — COMMENTED** [review]
Updated this with a warning, and some changes I found while manually testing. Works locally for me now. Still will need the new API field from https://github.com/openshift/api/pull/677
/hold

File: pkg/operator/configobservation/scheduler/observe_scheduler.go [hunk 2]
```diff
@@ -39,30 +40,73 @@ func ObserveSchedulerConfig(genericListers configobserver.Listers, recorder even
...
-		sourceTargetLocation,
-	)
+	if len(schedulerConfig.Spec.Profiles.Name) == 0 {
+		if len(currentProfiles) > 0 {
+			recorder.Eventf("ObservedConfigMapNameChanged", "scheduler profiles configmap removed")
+			unstructured.RemoveNestedField(prevObservedConfig, "profiles")
+		}
+		return prevObservedConfig, errs
+	}
+	profileConfig, err := listers.ConfigmapLister.ConfigMaps(operatorclient.GlobalUserSpecifiedConfigNamespace).Get(schedulerConfig.Spec.Profiles.Name)
```
Comment (ingvagabund) [inline pkg/operator/configobservation/scheduler/observe_scheduler.go:92]: `profileConfigMap`

File: pkg/operator/configobservation/scheduler/observe_scheduler.go [hunk 3]
```diff
@@ -39,30 +40,73 @@ func ObserveSchedulerConfig(genericListers configobserver.Listers, recorder even
...
+	if len(config) == 0 && len(currentProfiles) > 0 {
+		recorder.Eventf("ObservedProfilesChanged", "scheduler profiles removed")
+		return prevObservedConfig, errs
+	}
+	profiles := map[string][]interface{}{}
+	if err := yaml.Unmarshal([]byte(config), &profiles); err != nil {
+		errs = append(errs, err)
+		return prevObservedConfig, errs
+	}
+	if err := unstructured.SetNestedSlice(observedConfig, profiles["profiles"], "profiles"); err != nil {
```
Comment (ingvagabund) [inline pkg/operator/configobservation/scheduler/observe_scheduler.go:107]: Worth noting `observedConfig` here corresponds to `KubeSchedulerConfiguration` data type which is rendered into `config.yaml` filed passed to kube-scheduler binary.

**Review by damemi — COMMENTED** [review]
/close
we are taking a different direction with this, see the enhancement https://github.com/openshift/enhancements/blob/d40a31534debc62a5365aae052056e32db5ff449/enhancements/scheduling/scheduler-profiles.md

**Pull request closed without merging**